programmers-coding-test/sangjun/dfs-bfs/미로탐색.js

```diff
@@ -0,0 +1,51 @@
+const readline = require("readline");
+
+let input = [];
+let maps = [];
+
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout
+});
+
+rl.on("line", (line) => { input.push(line); }).on("close", () => { 
+    const [n, m] = input[0].split(" ").map(Number);
+    input.splice(0, 1);
+    for (let i = 0; i < n; i++) { 
+        maps.push(input[i].split("").map((el) => Number(el))); 
+    }   
```

Comment:
shift()로 접근하면 0번째꺼를 삭제함과 동시에 반환해주므로 아래처럼 하는게 가능함.
splice나 slice와 같은 메서드를 다시 호출할 필요가 없음.

이거 상준이 너 코드 어디에선가 본거 같은데.. 암튼 너 코드 중에 이렇게 한 거 있었음. 너꺼보고 좋은 것 같아서 내가 배우고 다시 너한테 알려준거임
```suggestion
    const [n, m] = input.shift().split(" ").map(Number);
    for (let i = 0; i < n; i++) { 
        maps.push(input.shift().split("").map((el) => Number(el))); 
    }   
```